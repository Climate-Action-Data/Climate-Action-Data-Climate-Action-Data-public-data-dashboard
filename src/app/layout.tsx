'use client'
import React from 'react'
import { Inter } from 'next/font/google'
import { Provider } from 'overmind-react'
import { config } from '../overmind'
const overmind = createOvermind(config)
import './globals.css'
import { createOvermind } from 'overmind'
import NoSSR from './noSSR'
import { CacheProvider } from '@chakra-ui/next-js'
import { ChakraProvider, extendTheme } from '@chakra-ui/react'
import { Header } from '../components/organisms/Header/Header'
import { brandPrimary, lightGray, lightGrayRound } from '@/styles/Button'
import { Menu } from '@/components/organisms/Menu/Menu'
import { mode } from '@chakra-ui/theme-tools'
import { cardSection } from '@/styles/Section'
import styles from './page.module.css'
import '../i18n'
/*eslint @typescript-eslint/quotes: off*/
const inter = Inter({ subsets: ['latin'] })

const theme = extendTheme({
  fontSizes: {
    xs: `10px`,
    sm: `12px`,
    md: `14px`,
    lg: `16px`,
    xl: `18px`,
    '2xl': `22px`,
    '3xl': `28px`,
    '4xl': `36px`,
    '5xl': `48px`,
    '6xl': `64px`,
  },
  components: {
    Skeleton: {
      baseStyle: {
        _light: {
          '--skeleton-start-color': `#717D81`,
          '--skeleton-end-color': `#949DA1`,
        },
        _dark: {
          '--skeleton-start-color': `#717D81`,
          '--skeleton-end-color': `#949DA1`,
        },
      },
    },
    Button: {
      variants: {
        brandPrimary,
        lightGrayRound,
        lightGray,
      },
    },
    Link: {
      variants: {
        brandPrimary,
      },
    },
    Container: {
      variants: {
        cardSection,
      },
    },
    Text: {
      baseStyle: {
        fontSize: `14px`,
      },
    },
  },
  colors: {
    lightGray: {
      main: `#00242C`,
      50: `#F8FAFA`,
      100: `#EDEFFE`,
      200: `#DBDEE0`,
      300: `#CACED0`,
      400: `#B8BEC0`,
      500: `#949DA1`,
      600: `#717D81`,
      700: `#4D5C62`,
      800: `#3E4A4E`,
      900: `#2E373B`,
    },
    gray: {
      main: `#00242C`,
      50: `#`,
      100: `#3E4A4E`,
      200: `#3E4A4E`,
      300: `#2E373B`,
      400: `#2E373B`,
      500: `#00242C`,
      600: `#00242C`,
      700: `#00242C`,
      800: `#00242C`,
      900: `#00242C`,
    },
    green: {
      main: `#2DEC7C`,
      50: `#EAFDF2`,
      100: `#D5FBE5`,
      200: `#2DEC7C`,
      300: `#2DEC7C`,
      400: `#2DEC7C`,
      500: `#2DEC7C`,
      600: `#24BD63`,
      700: `#1B8E4A`,
      800: `#125E32`,
      900: `#364D3F`,
    },
  },
  styles: {
    global: (props: any) => ({
      body: {
        bg: mode(`#F8FAFA`, `#000`)(props),
      },
    }),
  },
})

export default function RootLayout({ children }: { children: React.ReactNode }): React.JSX.Element {
  return (
    <Provider value={overmind}>
      <html lang="en">
        <body className={inter.className}>
          <NoSSR>
            <CacheProvider>
              <ChakraProvider theme={theme}>
                <Header />
                <Menu />
                <main className={styles.main}>{children}</main>
              </ChakraProvider>
            </CacheProvider>
          </NoSSR>
        </body>
      </html>
    </Provider>
  )
}
