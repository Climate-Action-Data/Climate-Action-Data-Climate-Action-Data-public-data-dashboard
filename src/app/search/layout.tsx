'use client'
import { SearchHeader } from '@/components/molecules/SearchHeader/SearchHeader'
import { Flex } from '@chakra-ui/react'

export default function SearchLayout({
  children, // will be a page or nested layout
}: {
  children: React.ReactNode
}) {
  return (
    <>
      <SearchHeader />
      <Flex>{children}</Flex>
    </>
  )
}
