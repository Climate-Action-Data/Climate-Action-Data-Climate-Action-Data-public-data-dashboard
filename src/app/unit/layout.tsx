'use client'

export default function UnitLayout({ children }: { children: React.ReactNode }) {
  return <>{children}</>
}
