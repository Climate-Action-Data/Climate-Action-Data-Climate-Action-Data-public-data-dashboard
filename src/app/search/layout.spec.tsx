import { render } from '@testing-library/react'
import layout from './layout'

it(`renders correctly`, () => {
  const { container } = render(layout({ children: <>test</> }))

  expect(container).toMatchSnapshot()
})
