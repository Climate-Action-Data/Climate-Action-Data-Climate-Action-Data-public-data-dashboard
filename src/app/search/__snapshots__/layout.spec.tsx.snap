// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-8y3igr"
    id="headerReference"
  >
    <div
      class="chakra-stack css-ryi8m5"
    >
      <div
        class="chakra-stack css-1jjq5p5"
      >
        <div
          class="css-0"
        >
          <h2
            class="chakra-heading css-0"
            style="margin-bottom: 16px; font-weight: 500;"
          />
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-input__group css-ulver4"
            data-group="true"
          >
            <input
              class="chakra-input css-klepor"
              data-testid="search-input-enter"
              type="text"
              value=""
            />
            <div
              class="chakra-input__right-element css-1qy9sq1"
            >
              <button
                class="chakra-button css-1z0w2s9"
                data-testid="search-header-button"
                type="button"
              >
                <div
                  class="css-70qvj9"
                >
                  Search
                  <svg
                    class="chakra-icon css-1hrwg97"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19.1623 20.2239L13.4084 14.4701C12.9084 14.8829 12.3334 15.206 11.6834 15.4393C11.0334 15.6726 10.361 15.7893 9.66613 15.7893C7.95696 15.7893 6.51045 15.1975 5.32658 14.0139C4.14271 12.8304 3.55078 11.3842 3.55078 9.67552C3.55078 7.96678 4.14256 6.52011 5.32613 5.33549C6.5097 4.15089 7.95585 3.55859 9.66458 3.55859C11.3733 3.55859 12.82 4.15053 14.0046 5.33439C15.1892 6.51826 15.7815 7.96478 15.7815 9.67394C15.7815 10.388 15.6616 11.0701 15.4219 11.7201C15.1821 12.3701 14.8623 12.9355 14.4623 13.4162L20.2411 19.1951C20.3796 19.3309 20.4488 19.4975 20.4488 19.695C20.4488 19.8925 20.3706 20.0695 20.214 20.226C20.0705 20.3695 19.8956 20.4412 19.6892 20.4412C19.4828 20.4412 19.3072 20.3688 19.1623 20.2239ZM9.66613 14.2893C10.9546 14.2893 12.0459 13.8422 12.9402 12.948C13.8344 12.0538 14.2815 10.9624 14.2815 9.67394C14.2815 8.38548 13.8344 7.29413 12.9402 6.39989C12.0459 5.50566 10.9546 5.05854 9.66613 5.05854C8.37766 5.05854 7.28631 5.50566 6.39208 6.39989C5.49786 7.29413 5.05076 8.38548 5.05076 9.67394C5.05076 10.9624 5.49786 12.0538 6.39208 12.948C7.28631 13.8422 8.37766 14.2893 9.66613 14.2893Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-k008qs"
  >
    test
  </div>
</div>
`;
